# 06-unlinkd.patch by Luigi Gangitano <luigi@debian.org>
#
# Only run unlinkd for diskd and ufs storage methods

--- src/main.c	2002-07-14 19:20:49.000000000 +0200
+++ src/main.c	2002-10-10 19:49:04.000000000 +0200
@@ -326,6 +326,21 @@
     asnFreeMemory();
 }
 
+#if USE_UNLINKD
+static int
+needUnlinkd(void)
+{
+    int i;
+    int r = 0;
+    for (i = 0; i < Config.cacheSwap.n_configured; i++) {
+	if (strcmp(Config.cacheSwap.swapDirs[i].type, "ufs") == 0 ||
+	strcmp(Config.cacheSwap.swapDirs[i].type, "diskd") == 0)
+	r++;
+    }
+    return r;
+}
+#endif
+
 static void
 mainReconfigure(void)
 {
@@ -350,6 +365,9 @@
 #endif
     redirectShutdown();
     authenticateShutdown();
+#if USE_UNLINKD
+    unlinkdClose();
+#endif
     externalAclShutdown();
     storeDirCloseSwapLogs();
     errorClean();
@@ -373,6 +391,9 @@
 #if USE_WCCP
     wccpInit();
 #endif
+#if USE_UNLINKD
+    if (needUnlinkd()) unlinkdInit();
+#endif
     serverConnectionsOpen();
     if (theOutIcpConnection >= 0) {
 	if (!Config2.Accel.on || Config.onoff.accel_with_proxy)
@@ -516,7 +537,7 @@
 
     if (!configured_once) {
 #if USE_UNLINKD
-	unlinkdInit();
+	if (needUnlinkd()) unlinkdInit();
 #endif
 	urlInitialize();
 	cachemgrInit();
